src/main/java/net/dries007/tfc/objects/items/metal/ItemProPick.java

```diff
@@ -0,0 +1,264 @@
+    public float getDestroySpeed(ItemStack stack, IBlockState state)
+    {
+        return canHarvestBlock(state, stack) ? efficiency : 1.0f;
+    }
+
+    /**
+     * Called when a Block is right-clicked with this Item
+     */
+    @Override
+    public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ)
+    {
+        IBlockState bState = worldIn.getBlockState(pos);
+
+        if (hand != EnumHand.MAIN_HAND || facing == null || !isValidProspectBlock(bState))
+            return super.onItemUse(player, worldIn, pos, hand, facing, hitX, hitY, hitZ);
+
+        // Play a block hit sound, reduce dirt block volume because its loud on default.
+        SoundType soundType = bState.getBlock().getSoundType(bState, worldIn, pos, player);
+        if (soundType.equals(SoundType.GROUND))
+            worldIn.playSound(player, pos, soundType.getHitSound(), SoundCategory.BLOCKS, soundType.getVolume()-0.5F, soundType.getPitch()-0.5F); //Think this sends to all except player
+        else
+            worldIn.playSound(player, pos, soundType.getHitSound(), SoundCategory.BLOCKS, soundType.getVolume(), soundType.getPitch()); //Think this sends to all except player
+
+        if (!worldIn.isRemote)
+        {
+            ItemStack itemStack = getOreBlock(bState);
+            if (itemStack != null)
+            {
+                //Current block is ore, tell result
+                player.sendStatusMessage(new TextComponentTranslation("tfc.propick.found").appendText(" " + itemStack.getDisplayName()),true);
+                player.getHeldItem(hand).damageItem(1, player);
+                setCooldown(player);
+            }
+            else
+            {
+                //Hit block 5 times to get a search result.
+                if (activeBlockPos == null || !activeBlockPos.equals(pos))
+                {
+                    activeBlockPos = pos;
+                    timesUsedOnBlock = 0;
+                }
+
+                timesUsedOnBlock += 1;
+                player.getHeldItem(hand).damageItem(1, player);
+
+                if (timesUsedOnBlock >= 5)
+                {
+                    activeBlockPos = null;
+                    timesUsedOnBlock = 0;
+
+                    scanSurroundingBlocks(worldIn, pos);
+
+                    if (results.isEmpty())
+                    {
+                        //Found nothing, tell nothing
+                        player.sendStatusMessage(new TextComponentTranslation("tfc.propick.found_nothing"),true);
+                        setCooldown(player);
+                    }
+                    else
+                    {
+                        ProspectResult[] list = results.values().toArray(new ProspectResult[0]);
+                        // todo: made show grades/multiple results if skilled prospector
+                        int index = results.size()== 1 ? 0 : random.nextInt(results.size()-1);
```

Comment:
Another why? This should display all results, not pick randomly.

Reply:
Nevermind, I'm forgetting how TFC works. But it should probably seed the random with the location. Try `random.setSeed(pos.toLong())`. That should allow us to have well defined false negatives that aren't bypassed by repeated clicking.